doc/tools/coqrst/coqdomain.py

```diff
@@ -400,57 +400,101 @@ class TableObject(NotationObject):
     def _name_from_signature(self, signature):
         return notation_to_string(signature)
 
+
 class ProductionObject(CoqObject):
     r"""A grammar production.
 
-    This is useful if you intend to document individual grammar productions.
-    Otherwise, use Sphinx's `production lists
+    Use prodn to document individual grammar productions instead of Sphinx
+    `production lists
     <http://www.sphinx-doc.org/en/stable/markup/para.html#directive-productionlist>`_.
 
-    Unlike ``.. productionlist``\ s, this directive accepts notation syntax.
-
-
-    Usage::
-
-       .. prodn:: token += production
-       .. prodn:: token ::= production
+    prodn displays multiple productions together with alignment similar to ``.. productionlist``,
+    i.e. displayed in 3 columns, however
+    unlike ``.. productionlist``\ s, this directive accepts notation syntax.
 
     Example::
 
-        .. prodn:: term += let: @pattern := @term in @term
         .. prodn:: occ_switch ::= { {? {| + | - } } {* @num } }
+        .. prodn:: term += let: @pattern := @term in @term
+        .. prodn:: | second_production
```

Comment:
Shouldn't this be a single `.. prodn::`, instead of one per line?

Reply:
I had thought of it as 3 separate examples.  With `.. prodn::` on each line, you get some extra space in the output.  I'll change to a single `.. prodn::`